src/Moryx.AbstractionLayer.Products.Endpoints/ProductConverter.cs

```diff
@@ -270,16 +270,21 @@ private void UpdateCollection(IList value, IEnumerable<PartModel> parts)
                     continue;
 
                 var match = unused.Find(r => r.Id == partModel?.Id);
+                // new partlink
                 if (match == null)
                 {
                     match = (IProductPartLink)Activator.CreateInstance(elemType);
+                    match.Product = _productManagement.LoadType(partModel.Product.Id);
                     value.Add(match);
                 }
+                //modified reference
+                else if (match.Product.Id != partModel.Product.Id)
+                    match.Product = _productManagement.LoadType(partModel.Product.Id);
                 else
+                 //part removed
                     unused.Remove(match);
```

Comment:
please reconsider name "unused": a removed part will be removed from the unused list and then we remove all remaining in the unused list to remove them from the value list? it making me headache to read this methode. ore maybe i do not unterstand it and it needs comments...